app/views/membership_applications/_membership_applications_list.html.haml

```diff
@@ -20,7 +22,7 @@
     %tbody
       - @membership_applications.each do |membership_application|
         %tr.applicant
-          %td.membership_number= membership_application.membership_number
+          %td.membership_number= link_to membership_application.membership_number, user_path(membership_application.user) unless membership_application.membership_number.blank?
```

Comment:
This works for sure - there is also a handy `link_to_unless` helper for this situation.